include/nbl/video/IAPIConnection.h

```diff
@@ -0,0 +1,38 @@
+#ifndef __NBL_I_API_CONNECTION_H_INCLUDED__
+#define __NBL_I_API_CONNECTION_H_INCLUDED__
+
+#include "nbl/core/IReferenceCounted.h"
+#include "nbl/video/surface/ISurfaceWin32.h"
+
+namespace nbl {
+namespace video
+{
+
+class IAPIConnection : public core::IReferenceCounted
+{
+public:
+    enum E_TYPE
+    {
+        ET_OPENGL,
+        ET_VULKAN
+    };
+
+    // TODO implement in some source file in src/nbl/...
+    static core::smart_refctd_ptr<IAPIConnection> create(E_TYPE apiType);
```

Comment:
not in the interface, in the derived class... i.e. `COpenGLConnection`

Reply:
i dont get it. What not in the interface?

Reply:
creation method shouldn't be a static in the interface... you should just have a 
```
COpenGLConnection::create
COpenGLESConnection::create
CVulkanConnection::create
CNVNConnection::create
CGNMConnection::create
```